### PR TITLE
moved getCircularReplacer() to telemetry-utils

### PR DESCRIPTION
**Pull request by chensixx**
Part of #8165 
moved getCircularReplacer() to telemetry-utils

### CODE REVIEW

File: common/lib/common-utils/src/parseErrorMessage.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        }
+        // Websocket errors reported by engine.io-client.
+        // They are Error objects with description containing WS error and description = "TransportError"
+        // Please see https://github.com/socketio/engine.io-client/blob/7245b80/lib/transport.ts#L44,
+        newMessage = `${newMessage}: ${JSON.stringify(error, getCircularReplacer())}`;
+    } else {
+        newMessage = `${message}: [object omitted]`;
+    }
+    return newMessage;
+}
```
Comment (markfields) [inline common/lib/common-utils/src/parseErrorMessage.ts:48]: I don't think this logic is general purpose enough to belong in common-utils.

Comment (markfields) [reply common/lib/common-utils/src/parseErrorMessage.ts:48]: Also I have plans to change it soon :P

File: packages/utils/telemetry-utils/src/errorLogging.ts [hunk 2]
```diff
@@ -259,6 +259,24 @@ function getValidTelemetryProps(obj: any, keysToOmit: Set<string>): ITelemetryPr
     return props;
 }
 
+// Borrowed from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Cyclic_object_value#examples
```
Comment (markfields) [inline packages/utils/telemetry-utils/src/errorLogging.ts:262]: Can you switch to doc comments `/** blah blah */`?

**Review by markfields — APPROVED** [review]
🎂 LGTM!  You might add/update a comment in the PR summarizing our conversation and the decision to go this route as opposed to common-utils, and also to refrain from lumping together the circular reference detection with pulling Error.message out.  For context/posterity.

**Comment by chensixx** [comment]
after conversation with Mark, we are just moving getCircularReference() out, to telemetry-utils.
1. getCircularReference is not perfect yet, as said in its comment, so might be better not moving it to common-utils for wide consumption.
2. we are not moving out the entire stringification logic because it's hard to have a generic way to handle any unknown error. Mark suggested adding types that reflect complexities of these errors, and use that to construct our errors with our desired granularity of info.

**Pull request merged**